astropy/time/core.py

```diff
@@ -1905,36 +1916,35 @@ def sidereal_time(self, kind, longitude=None, model=None):
             longitude = Longitude(longitude, u.degree,
                                   wrap_angle=180. * u.degree)
 
-        gst = self._erfa_sidereal_time(available_models[model.upper()])
+        theta = self._call_erfa(function, scales)
 
-        if model.upper() in ('IAU1982', 'IAU1994'):
-            # These models are pre-TIO, so just add the longitude
-            return Longitude(gst + longitude, u.hourangle)
-        else:
-            # These models include the TIO.
-            return Longitude(self._to_local(gst, longitude), u.hourangle)
-
-    if isinstance(sidereal_time.__doc__, str):
-        sidereal_time.__doc__ = sidereal_time.__doc__.format(
-            'apparent', sorted(SIDEREAL_TIME_MODELS['apparent'].keys()),
-            'mean', sorted(SIDEREAL_TIME_MODELS['mean'].keys()))
+        if include_tio:
+            # TODO: this duplicates part of coordinates.erfa_astrom.ErfaAstrom.apio;
+            # maybe posisble to factor out to one or the other.
+            sp = self._call_erfa(erfa.sp00, ('tt',))
+            xp, yp = get_polar_motion(self)
+            # Form the rotation matrix, CIRS to apparent [HA,Dec] at longitude 0.
+            r = (rotation_matrix(-yp, 'x', unit=u.radian)
+                 @ rotation_matrix(-xp, 'y', unit=u.radian)
+                 @ rotation_matrix(theta+sp, 'z', unit=u.radian))
 
-    def _erfa_sidereal_time(self, model):
-        """Calculate a sidereal time using a IAU precession/nutation model."""
+            # Solve for angle.
+            theta = np.arctan2(r[..., 0, 1], r[..., 0, 0])
```

Comment:
I don't think you can get away with this. In fact, I think that you will find that `theta` here is just the previous `theta+sp`. It's only after the final rotation by `longitude` that the local Earth rotation angle or sidereal time is `arctan2(r[0,1], r[0,0])`.

Reply:
This is due to the fact that the wobble matrix is non-Hermetian.

Reply:
I'm a bit confused: as it was originally, the very last matrix multiplication was an additional rotation by the longitude around the `z` axis. But all that does is add `longitude` to the angle calculated here. So, I thought it was faster and simpler to just add the angles afterwards.  Am I missing something.

Reply:
On second thought, it wouldn't work even if the wobble matrix was Hermitian.

Reply:
Damn, yes, I'm wrong. Trying to be too clever by half.

Reply:
I've been puzzling over this ever since SOFA made that change. Intuitively, it's because what happens is that the first rotation by `theta_sp` is around the CIP. Then the Earth is tilted and the rotation by longitude is around the offset ITRS pole, so the longitude isn't just added on. In fact, in the case where the only tilt is `Yp`, what you get is a spherical triangle with the sides being `theta_sp`, `longitude`, and `eral`.

Reply:
Thanks, that makes sense!